src/OrchardCore.Modules/OrchardCore.Admin/AdminAreaControllerRouteMapper.cs

```diff
@@ -9,33 +9,45 @@ namespace OrchardCore.Admin
 {
     public class AdminAreaControllerRouteMapper : IAreaControllerRouteMapper
     {
-        private readonly string _defaultAreaPattern;
+        private const string _defaultAreaPattern = "{area}/{controller}/{action}/{id?}";
+        private readonly string _adminUrlPrefix;
 
         public int Order => -1000;
 
         public AdminAreaControllerRouteMapper(IOptions<AdminOptions> adminOptions)
         {
-            _defaultAreaPattern = adminOptions.Value.AdminUrlPrefix + "/{area}/{controller}/{action}/{id?}";
+            _adminUrlPrefix = adminOptions.Value.AdminUrlPrefix;
         }
 
         public bool TryMapAreaControllerRoute(IEndpointRouteBuilder routes, ControllerActionDescriptor descriptor)
         {
-            if (descriptor.ControllerName == "Admin" ||
-                descriptor.ControllerTypeInfo.GetCustomAttribute<AdminAttribute>() != null ||
-                descriptor.MethodInfo.GetCustomAttribute<AdminAttribute>() != null
-                )
+            var controllerAttribute = descriptor.ControllerTypeInfo.GetCustomAttribute<AdminAttribute>();
+            var actionAttribute = descriptor.MethodInfo.GetCustomAttribute<AdminAttribute>();
+
+            if (descriptor.ControllerName != "Admin" && controllerAttribute == null && actionAttribute == null)
+            {
+                return false;
+            }
+
+            var pattern = _defaultAreaPattern;
+
+            if (!string.IsNullOrWhiteSpace(actionAttribute?.Template))
```

Comment:
Is it behaving the same way as [Route] applied on a controller (I believe as a prefix to all actions attributes).

Reply:
Yes, it's the same.

At first I wasn't sure how RouteAttribute works with controllers, so I've tested it:
```csharp
[Route("Sample")]
public class SampleController : Controller
{
    public IActionResult Index() => Content(nameof(Index));
    public IActionResult Other() => Content(nameof(Other));
}
```
If I try to navigate to `~/Sample/` I get this error because the same route gets applied to both actions:
![image](https://github.com/OrchardCMS/OrchardCore/assets/4007293/a1bc7745-cf52-4ee1-a198-cce50e816920)
And if I try to go to `~/Sample/Index` I get 404 error.

So it looks like the `[Route]` template is applied as-is, it doesn't become a prefix.

If I use `[Route("Sample/{action}")]` then `~/Sample/Index` and `~/Sample/Other` works. You can do the same with `[Admin]` too. I have included an advice to use the `{action}` in the template [here](https://github.com/OrchardCMS/OrchardCore/pull/15251/files#diff-0c3a28c7723c63ada3464e5065d04cdc7239c40b9539a5de5632b11ea87474fdR19).

Reply:
Turns out this is great for `AdminController`s, because you can simply add `[Admin("Users/{action}/{id?}")]` to the controller like [here](https://github.com/OrchardCMS/OrchardCore/pull/15251/files#diff-254176877b0597fca38954810afd6597eecc56718835a87df042721cc200c968R32).

Reply:
yes `[Admin("Users/{action}/{id?}")]` is great. Just be sure that it works with tenants that use prefix please. I think it';; still work. But just be sure it does. This should simplify lots of the boilerplate code we use in the startup file.

Reply:
Thankfully this works:

![image](https://github.com/OrchardCMS/OrchardCore/assets/4007293/ee609a56-5360-44f2-aaa6-0472ce9f587d)

![image](https://github.com/OrchardCMS/OrchardCore/assets/4007293/d66ad3b0-a9f2-4ea7-949e-4e59e1c3ece5)